### PR TITLE
add deprecation notice (tombstones) for all existing prow markdown files

### PR DESCRIPTION
**Pull request by listx**
This is part of https://github.com/kubernetes-sigs/prow/issues/4 to once-and-for-all give a facelift to all of our existing Prow docs. It does this by adding a clear deprecation notice to all markdown files under `/prow`, and linking to the (converted) snapshot in the docs repo. The converted snapshot was updated/adjusted in https://github.com/kubernetes-sigs/prow/pull/22, so that PR has to be merged first before we merge this in.

/hold

/cc @chaodaiG @alvaroaleman @cjwagner @BenTheElder @spiffxp

### CODE REVIEW

**Comment by chaodaiG** [comment]
should we also delete the old content to avoid divergence in the future?

**Comment by listx** [comment]
> should we also delete the old content to avoid divergence in the future?

I wanted to do this originally, but thought that this was too drastic a change (I assumed there would be some resistance to this --- but maybe I was wrong?). Regardless, we already have the blockade (automatic blocked paths hold, just as in this PR) set up so there's little risk in divergence.

We can also decide to delete the content in a future PR if we really want to, and I don't think this concern should block this PR.

**Comment by chaodaiG** [comment]
> Regardless, we already have the blockade (automatic blocked paths hold, just as in this PR) set up so there's little risk in divergence.

I might have got it wrong, but my initial impression was that the automatic blocked paths will be lifted once this migration is done. I can imagine that there is probably not need to update/delete READMEs, do we consider allowing adding READMEs in some directories to indicate their corresponding links on the docs site?

**Comment by listx** [comment]
> > Regardless, we already have the blockade (automatic blocked paths hold, just as in this PR) set up so there's little risk in divergence.
> 
> I might have got it wrong, but my initial impression was that the automatic blocked paths will be lifted once this migration is done.

I think the blockage should be in place for at least another year to educate contributors that we shouldn't be adding new documentation in /prow.

> I can imagine that there is probably not need to update/delete READMEs, do we consider allowing adding READMEs in some directories to indicate their corresponding links on the docs site?

This can be done on a case-by-case basis.

**Comment by spiffxp** [comment]
FWIW, I've seen us use more abrupt tombstones elsewhere in the project:
- delete the existing content, replace with link to new location
- add explicit date after which the tombstone will be deleted (could also link to issue)

**Comment by listx** [comment]
> FWIW, I've seen us use more abrupt tombstones elsewhere in the project:
> 
> * delete the existing content, replace with link to new location
> * add explicit date after which the tombstone will be deleted (could also link to issue)

Ah, good to know! This gives me the confidence to do the same in this PR. Unless someone votes against, I'll do the same here (expect an update from me in a few hours).

**Comment by listx** [comment]
Pushed up a new change to delete existing content. I've also arbitrarily set Nov 30 as the date when we'll delete these tombstones altogether.

File: prow/README.md [hunk 1]
```diff
@@ -1,124 +1,10 @@
...
+Please do not edit this file; instead, make changes to the new location!
 
-Prow is a Kubernetes based CI/CD system. Jobs can be triggered by various types of events and report their status to many different services. In addition to job execution, Prow provides GitHub automation in the form of policy enforcement, chat-ops via `/foo` style commands, and automatic PR merging.
+The new location is served on the web at
+https://docs.prow.k8s.io/docs/legacy-snapshot/.
 
-See the [GoDoc](https://godoc.org/k8s.io/test-infra/prow) for library docs.
-Please note that these libraries are intended for use by prow only, and we do
-not make any attempt to preserve backwards compatibility.
+This file will be deleted on 2022-11-30.
```
Comment (spiffxp) [inline prow/README.md:9]: For followup (maybe add this as a comment to this file): when you sweep through and delete `prow/**/*.md`, I would recommend keeping a README.md file here that explains "code here, docs there" with links to the other repo / website.

**Review by spiffxp — APPROVED** [review]
/approve
/lgtm
/hold cancel

**Pull request merged**